developer_tests/build_everything/run_all_quickbuilds.sh

```diff
@@ -4,77 +4,126 @@
 # by UCAR, "as is", without charge, subject to all terms of use at
 # http://www.image.ucar.edu/DAReS/DART/DART_download
 
-# options to set
-#   1. DART directory
-#   2. compiler
-
-export DART=$(git rev-parse --show-toplevel)
-FC=intel
+# Usage: run_all_quickbuilds.sh compiler [gcc intel nvhpc cce]
 
+if [ $# -eq 0 ]; then
+   echo "ERROR: expecting at least one argument"
+   exit 1
+fi
 
-if [[ ! -d $DART ]] ; then 
-  echo "No DART directory: " $DART
-  exit 1
-fi 
+compiler=$1
+branch=${2:-main} # checkout branch if given, or use main
 
-if [[ -z  $PBS_ENVIRONMENT ]]; then
-  echo "ERROR: Run this in a batch job"
+# Check if the script is running in a batch job or interactive session
+if [[ -z $PBS_ENVIRONMENT ]]; then
+  echo "ERROR: You must run this in a batch job"
   echo "       qsub submit_me.sh"
   echo "       or an interactive session"
   exit 2
 fi
 
-cp mkmf.template.compile_all $DART/build_templates/mkmf.template
-cd $DART
+# Specify the mkmf template for each compiler
+if [[ $compiler == "intel" ]]; then
+  mkmf_template="mkmf.template.intel.linux"
+elif [[ $compiler == "gcc" ]]; then
+  mkmf_template="mkmf.template.gfortran"
+elif [[ $compiler == "cce" ]]; then
+  mkmf_template="mkmf.template.cce"
+elif [[ $compiler == "nvhpc" ]]; then
+  mkmf_template="mkmf.template.nvhpc"
+else
+  echo "$compiler is not a valid argument"
+  exit 3
+fi
+  
 
-# run fixsystem 
-# running this once at the beginning otherwise all make commands will
-# try and alter the mpi_*_utilities_mod.f90 simultaneously
-cd assimilation_code/modules/utilities; ./fixsystem $FC
-cd -
+test_dir="/glade/derecho/scratch/$USER/build_everything/$compiler"
+if [[ -d $test_dir ]]; then
+  echo "Directory exists: $test_dir"
+  exit 4
+fi
+
+mkdir -p $test_dir
+cd $test_dir
+git clone 'https://github.com/NCAR/DART.git'
+cd DART
+export DART=$(git rev-parse --show-toplevel)
+git checkout $branch
+if [[ $? -ne 0 ]]; then
+   echo "unknown branch"
+   exit 5
+fi
+
+# mkmf for chosen compiler
+module load $compiler
+cp build_templates/$mkmf_template build_templates/mkmf.template
 
+# Run fixsystem to avoid all make commands from altering mpi_*_utilities_mod.f90 simultaneously
+cd assimilation_code/modules/utilities; ./fixsystem $compiler
```

Comment:
currently:
./fixsystem gcc

instead of gfortran, and compiling successfully 

GNU Fortran (GCC) 12.2.0 20220819 (HPE)

Reply:
fixed in [a20971a](https://github.com/NCAR/DART/pull/593/commits/a20971ad106b382c02056c6d269f57d50cfb423d)